src/libs/conduit/conduit_utils.cpp

```diff
@@ -932,6 +932,41 @@ check_num_char(const char v)
     return res;
 }
 
+//---------------------------------------------------------------------------//
+std::string
+strip_quoted_strings(const std::string &input, const std::string &quote_char)
+{
+    std::string res;
+    bool in_string = false;
+    bool emit = false;
+
+    for(size_t i = 0; i < input.size(); ++i)
+    {
+        emit = true;
+        // check for start & end of a string
+        if(input[i] == quote_char[0] && ( i > 0 && ( input[i-1] != '\\' )))
+        {
+            if(in_string)
+            {
+                in_string = false;
+                emit = false;
+            }
+            else
+            {
+                in_string = true;
+            }
```

Comment:
what is going on here?

"If it is in the string, then it is not in the string. If it is not in the string, then it is in the string."

Reply:
I guess this is saying that when we hit a quote_char, if we were in the quoted substring already then now we are exiting, and if we weren't then now we are entering. I would put comments to make it easier to understand.

Reply:
you only want to emit a character when you are not in a string, including the opening and closing quote.  This logic implements this. 

if you aren't in a string and you have encountered a quote, you are starting a string

If you are in a string and have encountered an ending quote, you are no longer in a string but you still don't want to emit the quote char.

Reply:
I know, it's just hard to read. I would take these words here and put them in the code.